subprojects/core/src/main/java/org/gradle/plugin/use/internal/DefaultPluginId.java

```diff
@@ -14,49 +14,51 @@
  * limitations under the License.
  */
 
-package org.gradle.plugin.internal;
+package org.gradle.plugin.use.internal;
 
 import com.google.common.base.CharMatcher;
 import org.gradle.api.Nullable;
+import org.gradle.plugin.internal.InvalidPluginIdException;
+import org.gradle.plugin.use.PluginId;
 
 import static com.google.common.base.CharMatcher.anyOf;
 import static com.google.common.base.CharMatcher.inRange;
 
-public class PluginId {
+public class DefaultPluginId implements PluginId {
```

Comment:
I talked with @alkemist a while ago and he suggested moving it to an interface so we can control the creation.